packages/ethers/src/router/get-message-and-data-by-tx-hash.ts

```diff
@@ -0,0 +1,71 @@
+import { AbstractProvider, LogDescription, Interface } from "ethers";
+import { EquitoMessage } from "@equito-sdk/core";
+import { routerAbi } from "@equito-sdk/evm";
+
+/**
+ * @title getMessageAndDataByTxHash
+ * @description Fetches and decodes the `MessageSendRequested` event from a transaction's logs using its hash.
+ * @dev This function retrieves the transaction receipt and parses the logs to extract the relevant event data.
+ * 
+ * @param {AbstractProvider} provider - The Ethers.js provider to interact with the blockchain.
+ * @param {string} routerContractAddress - The address of the router contract emitting the event.
+ * @param {string} txHash - The transaction hash for which the logs are to be retrieved and parsed.
+ * 
+ * @returns {Promise<{ message: EquitoMessage, messageData: string } | null>} 
+ * - Returns an object containing the decoded `message` and `messageData` if successful, otherwise throws an error.
+ * 
+ * @throws {Error} 
+ * - Throws an error if the transaction receipt is not found, if no logs match the contract address, or if log parsing fails.
+ * 
+ * @example
+ * const provider = ...;  // Your Ethers.js provider
+ * const routerContractAddress = '0x...';  // The router contract address
+ * const txHash = '0x...';  // The transaction hash
+ * 
+ * getMessageAndDataByTxHash(provider, routerContractAddress, txHash)
+ *     .then(result => {
+ *         if (result) {
+ *             console.log('Decoded Message:', result.message);
+ *             console.log('Decoded Message Data:', result.messageData);
+ *         }
+ *     })
+ *     .catch(error => {
+ *         console.error('Error:', error.message);
+ *     });
+ */
+export async function getMessageAndDataByTxHash(provider: AbstractProvider, routerContractAddress: string, txHash: string): Promise<{ message: EquitoMessage, messageData: string } | null> {
```

Comment:
```suggestion
export async function getMessageAndDataByTxHash(provider: AbstractProvider, routerContractAddress: string, txHash: string): Promise<{ message: EquitoMessage, messageData: string }> {
```

i see it never returns null cause you throw

Reply:
i have update to return null in case of no log